relayer/src/foreign_client.rs

```diff
@@ -942,7 +959,7 @@ impl<DstChain: ChainHandle, SrcChain: ChainHandle> ForeignClient<DstChain, SrcCh
 
     /// Returns a vector with a message for updating the client to height `target_height`.
     /// If the client already stores a consensus state for this height, returns an empty vector.
-    pub fn build_update_client_with_trusted(
+    pub fn wait_and_build_update_client_with_trusted(
```

Comment:
Maybe add a comment on what we are waiting for? Not your change but I think the use of `src_network_latest_...` here is misleading as it's `src_application_latest...`. 
And probably we will not be able to avoid to introduce `query_chain_latest..` vs `query_application_latest...`. 

Right now the code is confusing, just trying to figure out how much we can fix in this PR, or later in #2300 (where we will need to wait on chain) or yet another time.

Reply:
Well really what is meant here is `src_application_latest_height`, right? As in we want the application to have reached that height before continuing, since we'll be querying for client state and what not? So simply changing the name here should be fine?

I can also add a comment to specify what we're waiting for; unfortunately the name is long enough as it is, I don't think we can pack more information in there 😅

Reply:
[1807e7f](https://github.com/informalsystems/ibc-rs/pull/2304/commits/1807e7f96554bfbc212fe24e31d5f45bfcafa616)